demo/primefaces-users-portlet/src/main/java/com/liferay/faces/demos/service/CompanyLocalServiceTracker.java

```diff
@@ -25,7 +25,28 @@
  */
 public class CompanyLocalServiceTracker extends ServiceTracker<CompanyLocalService, CompanyLocalService> {
 
-	public CompanyLocalServiceTracker(BundleContext bundleContext) {
+	private BundleContext _bundleContext;
+	private CompanyLocalService _companyLocalService;
+	
+
+	public CompanyLocalServiceTracker(BundleContext bundleContext) {		
 		super(bundleContext, CompanyLocalService.class, null);
+		_bundleContext = bundleContext;
+	}
+
+	@Override
+	public CompanyLocalService addingService(ServiceReference<CompanyLocalService> reference) {
+		_companyLocalService = _bundleContext.getService(reference);
+		return _companyLocalService;
+	}
+
+	@Override
+	public void removedService(ServiceReference<CompanyLocalService> reference, CompanyLocalService service) {
+		_bundleContext.ungetService(reference);
+		_companyLocalService = null;
+	}
```

Comment:
@juangon This is exactly what `super.removedService` does.

Reply:
Actually it isn't. We are updating our variable so whenever we get it, it will be null, instead of a previous cached (stale) UserLocalService or CompanyLocalService.